plugins/utils/dashboard.nix

```diff
@@ -6,141 +6,378 @@
+
+            Using nixvim's raw type, `action` can be a function, e.g:
+
+            ```nix
+              action.__raw = "func(path) vim.cmd('Telescope find_files cwd=' .. path) end";
+            ```
+
+            If action is a string, it will be wrapped in a `<cmd>...<cr>` mapping.
+          '';
+
+      disable_move = helpers.defaultNullOpts.mkBool false ''
+        Disable movement keymaps in the dashboard buffer.
+      '';
+
+      shortcut_type =
+        helpers.defaultNullOpts.mkEnumFirstDefault
+          [
+            "letter"
+            "number"
+          ]
+          ''
+            The shortcut type.
+          '';
+
+      change_to_vcs_root = helpers.defaultNullOpts.mkBool false ''
+        When opening a file in hyper mru, it will change to the root of vcs.
+      '';
+
+      config = {
+        disable_move = helpers.defaultNullOpts.mkBool false ''
+          Disable movement keymaps in the dashboard buffer.
+        '';
+
+        week_header = {
+          enable = helpers.defaultNullOpts.mkBool false ''
+            Whether to use a header based on the current day of the week,
+            instead of the default "DASHBOARD" header.
+
+            A subheading showing the current time is also displayed.
+          '';
+
+          concat = helpers.defaultNullOpts.mkStr "" ''
+            Additional text to append at the end of the time line.
+          '';
+
+          append = helpers.defaultNullOpts.mkListOf types.str [ ] ''
+            Additional header lines to append after the the time line.
+          '';
+        };
+
+        header =
+          helpers.defaultNullOpts.mkListOf types.str
+            [
+              ""
+              " ██████╗  █████╗ ███████╗██╗  ██╗██████╗  ██████╗  █████╗ ██████╗ ██████╗  "
+              " ██╔══██╗██╔══██╗██╔════╝██║  ██║██╔══██╗██╔═══██╗██╔══██╗██╔══██╗██╔══██╗ "
+              " ██║  ██║███████║███████╗███████║██████╔╝██║   ██║███████║██████╔╝██║  ██║ "
+              " ██║  ██║██╔══██║╚════██║██╔══██║██╔══██╗██║   ██║██╔══██║██╔══██╗██║  ██║ "
+              " ██████╔╝██║  ██║███████║██║  ██║██████╔╝╚██████╔╝██║  ██║██║  ██║██████╔╝ "
+              " ╚═════╝ ╚═╝  ╚═╝╚══════╝╚═╝  ╚═╝╚═════╝  ╚═════╝ ╚═╝  ╚═╝╚═╝  ╚═╝╚═════╝  "
+              ""
+            ]
+            ''
+              The header text, displayed at the top of the buffer.
+            '';
+
+        footer = helpers.defaultNullOpts.mkListOf types.str [ ] ''
+          The footer text, displayed at the bottom of the buffer.
+        '';
+
+        # TODO add examples
+        shortcut = helpers.defaultNullOpts.mkListOf actionType [ ] ''
+          Shortcut actions to be added to the "hyper" theme.
+
+          Note: has no effect on the "doom" theme.
+        '';
+
+        # TODO add examples
+        center = helpers.defaultNullOpts.mkListOf actionType [ ] ''
+          Actions to be added to the center section of the "doom" theme.
+
+          Note: has no effect on the "hyper" theme.
```

Comment:
I think I'll add a warning when this is used with `cfg.settings.theme = "hyper"` (or `null`).

Although users might define both this _and_ `shortcut` then set `theme` conditionally...

Reply:
Don't think I'll do this.